src/anyp/Uri.cc

```diff
@@ -362,7 +362,11 @@ AnyP::Uri::parse(const HttpRequestMethod& method, const SBuf &rawUrl)
                 return false;
             *dst = '\0';
 
-            foundPort = scheme.defaultPort(); // may be reset later
+            // If the parsed scheme has no (known) default port, and there is no
+            // explicit port, then we will reject the zero port during foundPort
+            // validation, often resulting in a misleading 400/ERR_INVALID_URL.
+            // TODO: Remove this hack when switching to Tokenizer-based parsing.
+            foundPort = scheme.defaultPort().value_or(0); // may be reset later
```

Comment:
This diff may look as if this PR adds a "hack", but this PR does not change official code behavior. It only documents it. There is no bug here unless you consider a misleading `400/ERR_INVALID_URL` outcome a bug, and that misleading outcome problem existed before these changes.

Reply:
Side topic: do you have plans to do that conversion? or shall I resurrect my old project?

Reply:
I think the answer depends, in part, on what code you already have. Can you share a pointer to the corresponding git branch in your repository? To avoid misunderstanding, I am _not_ going to review that code -- I understand that it may be very stale and unpolished/unfinished -- I am only going to silently assess how far ahead it is and whether I agree with its overall direction, in order to answer your question.

Reply:
As far a I got was PR #335 to prepare the validation functions for using SBuf produced by a Tokenizer.

Reply:
Sure. All good.